core/src/main/java/hudson/model/Queue.java

```diff
@@ -799,8 +799,9 @@ private WaitingItem peek() {
     }
 
     private List<Item> filterItemListBasedOnPermissions(List<Item> r, Item t) {
-        if (t.task instanceof hudson.model.Item) {
-            if (((hudson.model.Item)t.task).hasPermission(hudson.model.Item.READ)) {
+        if (t.task instanceof hudson.security.AccessControlled) {
+            if (((hudson.security.AccessControlled)t.task).hasPermission(hudson.model.Item.READ)
+                    || ((hudson.security.AccessControlled) t.task).hasPermission(hudson.security.Permission.READ)) {
                 r.add(t);
             }
         }
```

Comment:
For what it is worth, it would seem reasonable to also add any that didn't implement `AccessControlled` as they don't have permissions *per se* but it seems safer to require such tasks to "step up" and implement `AccessControlled` if they actually want to be visible in the UI.